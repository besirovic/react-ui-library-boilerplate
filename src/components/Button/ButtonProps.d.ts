import * as React from "react";

export interface ButtonProps extends React.Props<Button> {
	className?: string;
}

export declare class Button extends React.Component<ButtonProps, any> {}
