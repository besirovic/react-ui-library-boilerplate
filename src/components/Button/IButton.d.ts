import * as React from 'react';

export interface ButtonProps {
  className?: string;
  children: React.ReactNode;
}

export function Button(props: ButtonProps): React.ReactNode;
