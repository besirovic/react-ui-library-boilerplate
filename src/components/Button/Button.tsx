import * as React from 'react';
import './style/button.scss';

const Button = () => <button>Click me</button>;

export default Button;
