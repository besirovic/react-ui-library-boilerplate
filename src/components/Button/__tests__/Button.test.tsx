import * as React from 'react';
import { shallow } from 'enzyme';

import Button from '../Button';

test('first test', () => {
  const wrapper = shallow(<Button>Default Button</Button>);

  expect(wrapper).toMatchSnapshot();
});
