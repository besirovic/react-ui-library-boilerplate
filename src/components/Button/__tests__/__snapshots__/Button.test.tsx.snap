// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`first test 1`] = `
<button>
  Click me
</button>
`;
